### DISCUSSION TITLE
Ring MQTT addon occasionally not starting correctly after HA restart

### DISCUSSION THREAD

**Discussion by snigehere** [tsightler/ring-mqtt#777] [General Discussion]
I have noticed in the past couple of months that occasionally, when I restart home assistant, the ring mqtt addon does not fully restart - There are messages in the log of the addon so some part is working but many entities are shown as unavailable .. I have waited some time to see if it was just downloading an update or something so running slower.  To solve it I just go to the addon, stop it and then restart it and it all comes up correctly.  

At the moment I have not investigated in any detail, and have not captured any logs etc, but wanted to see if anyone else is seeing this behaviour or if its just me.  If its just me I will dig in further to see what's going on - maybe a timing issue with other integrations starting up.  If others see it then I will capture some logs and more details to see if sharing info will help.

**Comment by snigehere** [comment]
I have had another couple of instances of this issue and have dumped the logs ... 

Its a bit odd .... Today I restarted home assistant installing an update and after the restart all of my ring devices (cameras and alarm) showed as unavailable.  After an hour or so I restarted the ring add on and it all came back up ok.

Ahead of restarting the add on I downloaded the ring add logs .. checking the log I don't see a restart at all that fits with the restart of home assistant - its like the add on kept running even though home assistant restarted.

My set up is virtual box under windows 10 with the broker and add on within the HA environment

I see in my home assistant log a restart of home assistant at 2024-02-09 11:54:35.059 but in the ring add on log there is no restart.

Maybe this is normal behaviour and something else is causing the devices to all go unavailable 

I will open a bug at some point but need to make sure I have all the details

I have the log from the last issue in case anyone wants to look at it ,,, as I have an alarm and 5 cameras its fairly busy.

**Reply by tsightler** [reply]
Restarting HA does not restart addons so that is not odd, it is expected.  Basically, there's HAOS, and then all HA components and addons run in their own containers.  When you restart HA, even the supervisor, only the HA container is restarted, the addon containers all stay running.  Only restarting the entire HAOS restarts everything.

To know when HA restarts, ring-mqtt monitors the MQTT birth/lastwill messages, see:

https://www.home-assistant.io/integrations/mqtt/#birth-and-last-will-messages

Basically, the addon monitors the homeassistant/status topic and, if it sees an "online" message it knows that HA was restarted and starts resending device discovery data (this event is visible in the ring-mqtt logs).  However, it only does this for 3 minutes, in the past I've seen cases, especially on systems with low memory, where HA sends the online state very early, but really isn't ready and so misses all the new discovery data.  Normally this happens on really lower powered devices, like an RPi3 or maybe even an RPi4 with low RAM or a super slow SD card.

**Comment by snigehere** [comment]
Thanks for the detail.. what should I look for in the logs to see it picks up the restart?

**Reply by tsightler** [reply]
Offline:
2024-02-09T17:19:01.367Z ring-mqtt Home Assistant state topic homeassistant/status received message: offline

Online:
2024-02-09T17:31:02.452Z ring-mqtt Home Assistant state topic homeassistant/status received message: online
2024-02-09T17:31:02.452Z ring-mqtt Home Assistant restart detected, resending device config/state in 15 seconds

**Comment by snigehere** [comment]
Ok - Thanks for your help and guidance .. its all a learning experience 
I see both the messages in the log
2024-02-09T11:54:22.131Z ring-mqtt Home Assistant state topic homeassistant/status received message: offline

continued messages from cameras and alarm followed by

2024-02-09T11:57:48.652Z ring-mqtt Home Assistant state topic homeassistant/status received message: online
2024-02-09T11:57:48.652Z ring-mqtt Home Assistant restart detected, resending device config/state in 15 seconds

there are then a few messages from sensors coming in - motion detection followed by

2024-02-09T11:58:03.659Z ring-disc Republishing existing device id: xxxxxxx

I then see republishing going on through a bunch of ID's ... it seems to loop 3 times through the same set
then I see the normal activity of movement being picked up and camera snapshots but when I looked in HA all of my alarm system sensors showed unavailable (I use the sensors to trigger other HA processes)

At this time I just have the log before I restarted the add on and a second log from when I did the restart
I didn't check the cameras so I need to dig deeper to gather as much information as I can when I next see this issue
